bots/bots/aggregation-bot.js

```diff
@@ -13,29 +17,53 @@ class AggregationBot {
    */
   async main() {
     const measurables = await this.api.measurables();
+    if (measurables.length === 0) {
+      console.log('Measurables list is empty.');
+      return true;
+    }
+
+    console.log(`Going to process ${measurables.length} measurables.`);
 
     for (const measurable of measurables) {
-      const measurableId = measurable.id;
-      const measurements = await this.api.measurementsCompetitive({
-        measurableId,
-      });
+      const id = { measurableId: measurable.id };
+
+      const measurements = await this.api.measurementsCompetitive(id);
+      console.log(`Going to process ${measurements.length} measurements.`);
+      if (measurements.length === 0) continue;
+
       const aggregated = await this.aggregate(measurements);
-      await this.api.measurementCreateAggregation({
-        measurableId,
-        ...aggregated,
-      });
+      if (!aggregated) continue;
+
+      await this.api.measurementCreateAggregation({ ...id, ...aggregated });
     }
 
     return true;
   }
 
   /**
+   * @todo: What to do with "floatPoint" spot?
    * @param {object[]} measurements
-   * @return {Promise<{floatPoint: number}>}
+   * @return {Promise<{floatPoint: number} | {floatCdf: {xs: number[], ys: number[]}} | null>}
    */
   async aggregate(measurements) {
+    const cdfs = measurements.filter((measurement) => {
+      return !!_.get(measurement, 'value.floatCdf');
+    }).map((measurement) => {
+      const xs = _.get(measurement, 'value.floatCdf.xs');
+      const ys = _.get(measurement, 'value.floatCdf.ys');
+      return new Cdf(xs, ys);
+    });
+
+    if (cdfs.length === 0) return null;
+
+    // @todo: What's about the limit?
+    const combined = new CdfCombination(cdfs).combine(10);
```

Comment:
The sample count should be 1000, but should probably also be a constant on the top of the file or config or similar

Reply:
Ok, will update

Reply:
Fixed.